### ISSUE TITLE
TypeAlias defined inside functions doesn't respect the typealias naming style.

### ISSUE THREAD

**Issue by yilei** [pylint-dev/pylint#8536]
### Bug description

Example code:

```python
# pylint: disable=missing-docstring
from typing import TypeAlias

GoodTopLevelName: TypeAlias = int
bad_top_level_name: TypeAlias = int


def my_func():
    GoodLocalName: TypeAlias = int
    bad_local_name : TypeAlias = int
    del GoodLocalName, bad_local_name
```

### Command used

```shell
pylint t.py
```

### Pylint output

```shell
t.py:5:0: C0103: Type alias name "bad_top_level_name" doesn't conform to predefined naming style (invalid-name)
```

### Expected behavior

It should also raise `invalid-name` for the `bad_local_name ` variable

### Pylint version

```shell
pylint 2.17.2
astroid 2.15.2
Python 3.11.2 (main, Mar  3 2023, 16:02:46) [GCC 12.2.0]
```

### Additional dependencies

I having a PR coming.

**Comment by Pierre-Sassoulas** [comment]
Thank you for reporting and working on this. Is it a regression from pylint 2.17.2 ?

**Comment by yilei** [comment]
Not a regression, but a missed code path in https://github.com/pylint-dev/pylint/pull/7116
